addons/delivery/models/sale_order.py

```diff
@@ -25,6 +34,24 @@ def _compute_amount_total_without_delivery(self):
         delivery_cost = sum([l.price_total for l in self.order_line if l.is_delivery])
         return self.amount_total - delivery_cost
 
+    @api.depends('carrier_id')
+    def _compute_shipping_method(self):
```

Comment:
Should we really need a compute for it?

Reply:
I.M.O it's not because the carrier allow to select a pickup point that we have to define it as a default

Reply:
As far as I understand, if the delivery carrier has `<carrier>_use_locations` _and_ if it is true at the same time, then it's always used as a pickup delivery. Isn't this what happens [here](https://github.com/odoo-dev/odoo/blob/8d10ed6942f2f8c9429355d3e09f17d784745917/addons/website_sale/views/website_sale_delivery_templates.xml#L25-L38) or am I missing something?